pkg/manager-nnf/storage_pool.go

```diff
@@ -111,24 +112,33 @@ func (p *StoragePool) findStorageGroupByEndpoint(endpoint *Endpoint) *StorageGro
 
 func (p *StoragePool) recoverVolumes(volumes []storagePoolPersistentVolumeInfo, ignoreErrors bool) error {
 
-	for _, volume := range volumes {
-		storage := p.storageService.findStorage(volume.SerialNumber)
-
-		if storage != nil {
-			volume, err := storage.FindVolumeByNamespaceId(volume.NamespaceId)
-			if err != nil {
-				if ignoreErrors {
-					continue
-				}
-				
-				return err
+	log.Infof("Storage Pool %s: Recover Volumes", p.id)
+
+	for _, volumeInfo := range volumes {
+
+		// Locate the NVMe Storage device by Serial Number
+		storage := p.storageService.findStorage(volumeInfo.SerialNumber)
+		if storage == nil {
+			log.Warnf("Storage %s not found", volumeInfo.SerialNumber)
+			continue
```

Comment:
Not finding storage could be because a drive dropped offline, and I think this is another open TODO where the storage pool should report Critical failure. @bdevcich-hpe we should talk about this and the other TODO below.